handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -306,6 +306,16 @@ public void operationComplete(Future<SslContext> future) throws Exception {
     }
 
     /**
+     * The default implementation will simply call {@link AsyncMapping#map(Object, Promise)} but
+     * users can override this method to implement custom behavior.
+     *
+     * @see AsyncMapping#map(Object, Promise)
+     */
+    protected Future<SslContext> lookup(ChannelHandlerContext ctx, String hostname) throws Exception {
```

Comment:
I should have read this before I made https://github.com/netty/netty/pull/5786#discussion_r77205752, but it seems unnecessary to expose the `mapping` member variable when users can just call `super.lookup`.